### PR TITLE
Death of Superhero, Removal of Theme Builder, and Addition of New Themes

### LINKED ISSUE #3799: Superhero Theme is Dead
The Superhero theme is something that I regret creating as it has been a glitchy mess that I haven't been able to resolve. That is why I will be removing the Superhero theme as well as the Bootswatch dependency that it relied upon. This is a sad state of affairs yet I will try to find another theme to replace it in the coming weeks.

### LINKED ISSUE #3838: BS4 font size.
### Describe the bug
Not a bug per se but, it bugs me!
I have no idea why the authors at twitter decided to increase the default font size from 14 to 16px but, for a business application, 14px is more appropriate.

Remembering again, we implement BS mainly to ease responsiveness and styling uniformity . Doesn't mean we are bound by all that is bootstrap.

Having sprawling UI's is annoying with as many interfaces presented to the user as possible, not scrolling and requiring as few interactions(accept business logic) as possible.
This also includes large titles with overspacing.

Many of our UI's originally were designed with this in mind. This is why we have so many inling stylings etc and trying to achieve uniformity is very desirable however, these large UIs are counterintuitive imo. Sometimes we just have to bite the bullet and redesign some of our UI templates and not try to style around that templates original intent. Many times this requires converting the use of using the old pattern of using tables for layouts.

Originally my vision for BS4 was to have a default BS that had adjusted margins, padding and fonts to be more appropriate to our application.

I don't mind the push back but, for me, I can't seem to let this go.:)

### PR DESCRIPTION
**Pull request by tywrenn**
Fixes #3799 
Fixes #3838
Fixes #2059 

#### Short description of what this resolves:

- Removes the superhero theme
- Fixes a color issue with a calendar icon.

#### To Do:

- [x] Remove Theme Builder
- [x] Replace Theme Builder with compact mode option
- [x] Allow compact mode to be used on any theme

### CODE REVIEW

File: interface/themes/core.scss [hunk 1]
```diff
@@ -12,9 +12,6 @@
 @import "../../public/assets/@ttskch/select2-bootstrap4-theme/src/layout";
 
 @import "directional";
-
-@import url("themeBuilder.php"); // reads from user preferences and overrides fonts
```
Comment (bradymiller) [inline interface/themes/core.scss:16]: whooohooooo, to the fire! :)

tenor (4)

**Comment by tywrenn** [comment]
Changelog:
- Say hello to the new Solar theme! This will replace Superhero.
- Adds favicon to all pages on core headers. This should fix #2059 
- Updates Bootswatch to 4.5.2 (yeah i'm still using this)
- Fixes an issue where the colored themes aren't using the correct font size
- Consolidated and fixed a lot of colored theme issues
- Adds escape characters on default tags for core headers
- Adds escape character to stylesheet tag
- Other fixes

**Comment by tywrenn** [comment]
@bradymiller Ready for review and merging

File: gulpfile.js [hunk 2]
```diff
@@ -74,22 +74,37 @@ const autoGeneratedHeader = `
...
+        .pipe(postcss([prefix()]))
+        .pipe(gap.prependText(autoGeneratedHeader))
+        .pipe(gulpif(!config.dev, csso()))
+        .pipe(gulpif(!config.dev, sourcemaps.write()))
+        .pipe(gulp.dest(config.dest.themes));
 }
 // standard themes css compilation
 function styles_style_uni() {
+    // First, create the compact version
+    gulp.src(config.src.styles.style_uni)
```
Comment (bradymiller) [inline gulpfile.js:89]: this is increasing build time about 2.5x . Just making sure these new commands are async (I am assuming they are, just want to verify).

**Comment by bradymiller** [comment]
Very nice on testing. Long build time, but not sure that can be helped. Also a couple travis things to fix:
https://travis-ci.org/github/openemr/openemr/builds/720490157
(travis is still doing testing, just not letting us know on github)

File: gulpfile.js (hunk 2 above)
Comment (tywrenn) [reply gulpfile.js:89]: This is async as each theme file creates a compact version followed by a regular version. This created another issue for me which is the fact that creating a compact version of each one is basically the same as compiling all the themes one more time therefore increasing compile time. I'll have to find a more elegant solution for this but for now the build time will be a bit long until I find another solution.

**Comment by tywrenn** [comment]
Darn the ghost of Travis CI gets me every time 😂😂

File: gulpfile.js (hunk 2 above)
Comment (tywrenn) [reply gulpfile.js:89]: Or at least I think it's async...

Comment (bradymiller) [reply gulpfile.js:89]: i was able to get minimal improvement (from 4:30 to 4:10) if break these into their own functions/calls. I'll do some more testing on that and plan to bring it in after this PR (will bring this in after silent travis issues are fixed :) ).

**Comment by tywrenn** [comment]
@bradymiller Fixed Travis

**Comment by bradymiller** [comment]
hi @tywrenn , Bringing this in (have a commit following this that decreases build time by about 5-7%). Looking forward to see if this blows up the demo farm :)

**Comment by bradymiller** [comment]
btw, thanks for the improvements! @sjpadgett will be very happy to see the compact mode.

**Comment by tywrenn** [comment]
Hi Solar theme 🙂 looking a bit out-of-this-world lol

**Comment by bradymiller** [comment]
just brought in a commit that salvages a little bit of time on the build (not sure why it does):
https://github.com/openemr/openemr/commit/d6111ce6e40268662ac99b06e8c3545ffd1728b6

**Comment by bradymiller** [comment]
yep, Solar themes will be coming to demos very soon (in 4 or so hours assuming the cpu build doubling doesn't blow up the farm :smiley:  ) : 
https://github.com/openemr/demo_farm_openemr/commit/be5f769292428bff080b51a4058fb9b70f1bd468

**Comment by sjpadgett** [comment]
> btw, thanks for the improvements! @sjpadgett will be very happy to see the compact mode.

I'm not very happy, i'm ecstatic! Not only with compact but, the login looks great. Login to right is my favorite but, centered is up there. I did note some minor issues on darker themes where some nav items are wrong color or not showing. Minor stuff.

I know this stuff may seem trivial to many of you but, my past experiences with users and their UX will make the difference between liking an application or just having a ho hum experience. Regardless how well the application performs. For health care applications, I strongly feel compact is important.

@tywrenn , very nice job pulling this off. I'd wait ten minutes to build!:) Happy you were able to lose themebuilder because believe it or not, it was a bottleneck in performance(having to require globals). 

I also feel we can further trim some themes from codebase. Say by half. Maybe get rid of the ones that are close in color. Seriously, let's push to do that!

Again Tyler, i'm very impressed and thank you so much for humoring this old man!:) Great job!

**Pull request merged**